src/feasibility-form-nls.jl

```diff
@@ -259,7 +259,7 @@ function NLPModels.hess(nlp::FeasibilityFormNLS, xr::AbstractVector; obj_weight:
   @lencheck nlp.meta.nvar xr
   increment!(nlp, :neval_hess)
   n, ne = nlp.internal.meta.nvar, nlp.internal.nls_meta.nequ
-  return [spzeros(n, n + ne); spzeros(ne, n) obj_weight*I]
+  return Symmetric([spzeros(n, n + ne); spzeros(ne, n) obj_weight*I], :L)
```

Comment:
This is more a comment than a review. If I understand well the spirit behind NLPModels 0.16 is to force the return type to be a `Symmetric` and not just an economical way of returning the complete matrix?

Because, in this function, for instance, the returned matrix was already symmetric, so it is slightly strange and maybe a bit restrictive to force the `Symmetric`?

Reply:
I'll remove it.

Reply:
There's also all the other `hess` methods. Shouldn't they be removed as well?

Reply:
FeasibilityResidual only implements `hess`, because it has `Jx'*Jx` in there.
SlackModel's `hess` was kept in case the underlying model has a specialized `hess`, or didn't implement `hess_coord`. Currently is not tested, as far as I could tell. Should I remove it?

Reply:
I forget why we decided not to implement `hess_coord` instead (by computing Jx'Jx)? Do you remember?

Reply:
Because of the sparse case. To implement `hess_coord` we need to compute `Jx' * Jx` when creating the model to define `nnzh`, `rows` and `cols`.